app/views/proposals/query.html.haml

```diff
@@ -2,16 +2,57 @@
   = render partial: 'shared/search_ui'
 
 .inset
-  %h1
-    Proposals participated in
-  %strong Currently displaying results for:
-  = property_to_s(@text)
-  - if @adv_search.to_s.present?
-    = @adv_search.to_s
+  - page_title = @report ? @report.name : "Proposals participated in"
+  %h1 #{page_title}
+  - if @proposals_data.es_response
+    %strong #{@proposals_data.es_response.results.total} results for:
+  - else
+    %strong Currently viewing results for:
+  - if @search_query["humanized"].present?
+    %pre
+      = @search_query["humanized"]
   %br
   %a{ :href => proposals_path } Clear search terms
+  #query-links.pull-right
+    %ul.list-inline
+      %li
+        %a{ href: "#{download_proposals_path}?#{request.query_parameters.to_query}" }
+          Download
+          %i.fa.fa-download
+      %li
+        %a{ href: "#", onclick: '$("#save-search").modal()' }
+          Save as Report
+          %i.fa.fa-save
   = datespan_header(@start_date, @end_date)
-  = render partial: 'shared/table', locals: {container: @proposals_data}
+  - if @proposals_data.rows.any?
+    = render partial: 'shared/table', locals: {container: @proposals_data}
+    - if @proposals_data.es_response
+      #pages
+        = paginate @proposals_data.es_response, current_page: (params[:page]||1).to_i
   - if @proposals_data.rows.empty?
-    %p.empty-list-label
+    .empty-list-label
       No results
+
+:javascript
+  // global var referenced by search.js
+  var C2searchQuery = #{@search_query.to_json}
```

Comment:
I initial-capped it to indicate it is a global var. Happy to use whatever syntax best communicates that. Some folks prepend with a `g` -- I was using `C2`.